Apps/Database/src/Delegates/IEmailDelegate.cs

```diff
@@ -54,8 +54,9 @@ public interface IEmailDelegate
         /// Inserts an email using a populated Email object.
         /// </summary>
         /// <param name="email">The populated email to save.</param>
+        /// <param name="commit">If true, the record will be written to the DB immediately.</param>
         /// <returns>Returns the guid of the saved email.</returns>
-        Guid InsertEmail(Email email);
+        Guid InsertEmail(Email email, bool commit = true);
```

Comment:
nit: commit reads more of a commit object than a flag. How about **shouldCommit** or **isPartOfTransaction**?

Reply:
I've changed the reference to shouldCommit